### PR TITLE
refactor: refactor ProposeId implement

### PR DESCRIPTION
**Pull request by Phoenix500526**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
At present, the definition of `ProposeId` is `struct ProposeId(String)`. The reason for using a tuple struct is to implement some methods for the `ProposeId`, such as `new` and `into_inner`. However, this is unnecessary since the traits derived by `ProposeId` are already implemented by `String`, and `String` also provides a `from` method that can replace the `new` method. Therefore, defining `ProposeId` as an alias for `String` can make `ProposeId` more user-friendly. Additionally, whether `ProposeId` is defined as a type alias or as a tuple struct, the amount of work required to change the `ProposeId` type is roughly the same because all the call sites for the `new` method would need to be modified.

Furthermore, many business servers currently define their own `generate_propose_id`, but their implementations are identical. Therefore, I have extracted a common `generate_propose_id` method to reduce code duplication

* what changes does this pull request make?
1. modify ProposeId as a type alias
2. extract a shared function `generate_propose_id` to reuse

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

**Review by iGxnon — DISMISSED** [review]

File: curp-external-api/src/cmd.rs [hunk 1]
```diff
@@ -1,9 +1,10 @@
-use std::{fmt::Display, hash::Hash};
+use std::hash::Hash;
 
 use async_trait::async_trait;
 use engine::Snapshot;
 use prost::DecodeError;
-use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use serde::{de::DeserializeOwned, Serialize};
+use uuid::Uuid;
```
Comment (themanforfree) [inline curp-external-api/src/cmd.rs:7]: `curp` and `xline-client` no longer need to depend on `uuid`

File: curp/src/server/gc.rs [hunk 2]
```diff
@@ -96,48 +96,48 @@ mod tests {
...
+            .insert(ProposeId::from("3"), Ok(0.into()));
 
         // at 1100ms, the first two kv should be removed
         tokio::time::sleep(Duration::from_millis(500)).await;
         let board = board.write();
         assert_eq!(board.er_buffer.len(), 1);
         assert_eq!(
             board.er_buffer.get_index(0).unwrap().0,
-            &ProposeId::new("3".to_owned())
+            &ProposeId::from("3")
```
[the remark this answers is not in the document]
Comment (Phoenix500526) [reply curp/src/server/gc.rs:135]: Done

File: curp/src/server/gc.rs [hunk 3]
```diff
@@ -96,48 +96,48 @@ mod tests {
...
         );
         assert_eq!(board.asr_buffer.len(), 1);
         assert_eq!(
             board.asr_buffer.get_index(0).unwrap().0,
-            &ProposeId::new("3".to_owned())
+            &ProposeId::from("3")
```
[the remark this answers is not in the document]
Comment (Phoenix500526) [reply curp/src/server/gc.rs:140]: Done

File: curp/src/rpc/mod.rs [hunk 4]
```diff
@@ -199,13 +198,13 @@ impl WaitSyncedRequest {
...
-            propose_id: id.clone().into_inner(),
+            propose_id: id.clone(),
         }
     }
 
     /// Get the propose id
     pub(crate) fn propose_id(&self) -> ProposeId {
-        ProposeId::new(self.propose_id.clone())
+        self.propose_id.clone()
     }
```
[the remark this answers is not in the document]
Comment (Phoenix500526) [reply curp/src/rpc/mod.rs:208]: Done

File: curp/src/rpc/mod.rs [hunk 5]
```diff
@@ -199,13 +198,13 @@ impl WaitSyncedRequest {
     /// Create a `WaitSynced` request
     pub(crate) fn new(id: &ProposeId) -> Self {
         Self {
-            propose_id: id.clone().into_inner(),
+            propose_id: id.clone(),
         }
     }
```
[the remark this answers is not in the document]
Comment (Phoenix500526) [reply curp/src/rpc/mod.rs:203]: Done

File: curp-external-api/src/cmd.rs (hunk 1 above)
Comment (Phoenix500526) [reply curp-external-api/src/cmd.rs:7]: Removed

**Review by themanforfree — APPROVED** [review]

**Review by iGxnon — APPROVED** [review]

**Pull request merged**